### DISCUSSION TITLE
Unable to install SSCHA on Mac - ERROR: Could not build wheels for python-sscha

### DISCUSSION THREAD

**Discussion by halbuhai** [SSCHAcode/python-sscha#185] [Q&A]
Hi,

I'm trying to install SSCHA on my MacOS (Intel-based) using Anaconda. I have installed all dependencies as instructed in https://sscha.eu/download/. However, when I enter "pip install cellconstructor python-sscha tdscha" in the terminal, I get an error message that says:

 "Building wheel for python-sscha (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [55 lines of output]"
  
 and then, after many lines in the terminal, I get:
 
 "note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed cleaning build dir for python-sscha
Failed to build python-sscha
ERROR: Could not build wheels for python-sscha, which is required to install pyproject.toml-based projects"

Notes:
- I have the latest version of pip, setuptools, and wheel
- I have Python version 3.10.2; I am not sure if it's compatible with SSCHA 1.4

I appreciate your support.
Hassan

**Comment by mesonepigreco** [comment]
Hi, this is strange. Can you redirect the output and errors of the installation to a file and attach them?

Using something like:
```bash
pip install python-sscha > output_install_scha.txt 2> error_install_scha.txt
```
Then, attach the two files to this discussion so we know exactly what failed.

Yes, python 3.10 is fully supported, as is 3.11 (3.12 not yet)

**Reply by halbuhai** [reply]
Thank you for your reply. The two files are attached.
[error_install_scha.txt](https://github.com/SSCHAcode/python-sscha/files/14073448/error_install_scha.txt)
[output_install_scha.txt](https://github.com/SSCHAcode/python-sscha/files/14073449/output_install_scha.txt)

**Reply by mesonepigreco** [reply]
Mmm... the error is strange. It seems it is conflicting with a previous installation.

Try to download the version from this repository (both CellConstructor and python-sscha). Unpack it, go with the terminal to that directory

Then, inside the anaconda enviroment, instead of using the pip install command, type


```
python setup.py install >output.txt 2>errors.txt
```

And copy those two files.

Do it for cellconstructor first to understand if there is a problem with the compilation or with the python installation.

**Reply by halbuhai** [reply]
Here are the two files:
[errors.txt](https://github.com/SSCHAcode/python-sscha/files/14077310/errors.txt)
[output.txt](https://github.com/SSCHAcode/python-sscha/files/14077311/output.txt)

**Reply by diegomartinez2** [reply]
This is strange, most of this errors came from " warning: misleading indentation" in symphmodule.c,  secondordermodule.c, thirdordermodule.c, etc. 
Are all this coming from F2py (Fortran to Python interface generator)?

**Reply by diegomartinez2** [reply]
Correction: Those are warnings and they are not the cause of the compilation error. I think that this is the cellconstructor part that compiles (with warnings, but not errors).

**Comment by elchatz** [comment]
Hello,

Is there any updates on this, because I am having a similar problem in Windows and Linux, with Python >=3.11. I have installed cellconstructor 1.4.1 and tdsscha 1.4.0 in Linux with Python 3.6.3 and setuptools 36.5.0 but it is not running due to old version of numpy.

I am almost convinced that the problem is with the numpy.distutils.

Can you tell me a working version configuration to install the modules?

**Reply by mesonepigreco** [reply]
Hi, you should use python <= 3.11 (3.9 
to 3.11 should work), as newver version of numpy required by python 3.12 and on dropped support for autocompiling Fortran code which is shipped with the code. 

There is a nice installation command using anaconda on linux (and i believe windows) in the install page of the website www.sscha.eu. In this way you may have the specific libraries version to install the code correctly

**Reply by elchatz** [reply]
Since I am using conda, you probably mean 

`conda create -n sscha -c conda-forge python=3.11 gfortran libblas lapack openmpi julia openmpi-mpicc pip numpy scipy spglib=2.2`

This already installs Python 3.11.10, but I tried forcing 3.11.0 and the problem is the same:

`6d93128a3da9b0d313/FModules/interp.f90:95: undefined reference to dgetrs_'`

julia is only installed from pip and openmpi-mpicc is not found at all (on windows).
